File: android/SceneWorld001/app/src/main/java/z3d/display/Display3D.java
package z3d.display;

import z3d.base.Object3D;
import z3d.scene.Scene3D;
import z3d.vo.Matrix3D;
import z3d.vo.Vector2D;
import z3d.vo.Vector3D;

public class Display3D<bool> extends Object3D {
    public Scene3D scene3d;
    public Matrix3D posMatrix3d;
    public  Display3D(Scene3D val  ){
        this.scene3d=val;
        this.posMatrix3d=new Matrix3D();
    }

    public boolean getCanRender() {
        return true;
    }
    public void upData()
    {
    }
    public void  updateMatrix()
    {
        this.posMatrix3d.identity();
        this.posMatrix3d.appendScale(this.scaleX, this.scaleY, this.scaleZ);
//        this.posMatrix3d.appendRotation(this.rotationX, Vector3D.X_AXIS);
//        this.posMatrix3d.appendRotation(this.rotationY, Vector3D.Y_AXIS);
//        this.posMatrix3d.appendRotation(this.rotationZ, Vector3D.Z_AXIS);
//        this.posMatrix3d.appendTranslation(this.x, this.y, this.z);
        this.posMatrix3d.appendTranslation(0, 0,0);

    }

}

File: metalios/iosgl/iosgl/bt/display/display3d/Display3D.m
//
//  Display3D.m
//  iosgl
//
//  Created by zhao on 7/12/2019.
//  Copyright © 2019 zhao. All rights reserved.
//
#import "Scene3D.h"
#import "Display3D.h"
#import "Matrix3D.h"
#import "Scene3D.h"

@implementation Display3D
- (instancetype)init
{
    self = [super init];
    if (self) {
        self.posMatrix3d=[[Matrix3D alloc]init];
        self.modeMatrix=[[Matrix3D alloc]init];
        self.rotationMatrix3D=[[Matrix3D alloc]init];
    }
    return self;
}
- (instancetype)init:(Scene3D*)val;
{
    self = [super init];
    if (self) {
        self.scene3d=val;
        self.posMatrix3d=[[Matrix3D alloc]init];
        self.modeMatrix=[[Matrix3D alloc]init];
        self.rotationMatrix3D=[[Matrix3D alloc]init];
    }
    return self;
}
-(void)updateMatrix;
{
 
    [self.posMatrix3d identity];
    [self.posMatrix3d appendScale:_scaleX y:_scaleY z:_scaleZ];
    [self.posMatrix3d appendRotation:_rotationX axis:Vector3D.X_AXIS];
    [self.posMatrix3d appendRotation:_rotationY axis:Vector3D.Y_AXIS];
    [self.posMatrix3d appendRotation:_rotationZ axis:Vector3D.Z_AXIS];
    [self.posMatrix3d appendTranslation:_x y: _y z:_z];
    
    
    [self.rotationMatrix3D identity];
    [self.rotationMatrix3D appendRotation:_rotationX axis:Vector3D.X_AXIS];
    [self.rotationMatrix3D appendRotation:_rotationY axis:Vector3D.Y_AXIS];
    [self.rotationMatrix3D appendRotation:_rotationZ axis:Vector3D.Z_AXIS];
 

}
 
-(Matrix3D*)viewMatrix;
{
    if(self.scene3d){
        return self.scene3d.camera3D.modelMatrix;
    }else{
        return nil;
    }
}
-(void) upFrame  ;{
}
-(void) destory ;{
    
}
 

 
@end
